homeassistant/components/subaru/sensor.py

```diff
@@ -0,0 +1,251 @@
+    },
+    {
+        SENSOR_TYPE: "Tire Pressure FR",
+        SENSOR_FIELD: sc.TIRE_PRESSURE_FR,
+        SENSOR_UNITS: PRESSURE_HPA,
+    },
+    {
+        SENSOR_TYPE: "Tire Pressure RL",
+        SENSOR_FIELD: sc.TIRE_PRESSURE_RL,
+        SENSOR_UNITS: PRESSURE_HPA,
+    },
+    {
+        SENSOR_TYPE: "Tire Pressure RR",
+        SENSOR_FIELD: sc.TIRE_PRESSURE_RR,
+        SENSOR_UNITS: PRESSURE_HPA,
+    },
+    {
+        SENSOR_TYPE: "External Temp",
+        SENSOR_FIELD: sc.EXTERNAL_TEMP,
+        SENSOR_UNITS: TEMP_CELSIUS,
+    },
+    {
+        SENSOR_TYPE: "12V Battery Voltage",
+        SENSOR_FIELD: sc.BATTERY_VOLTAGE,
+        SENSOR_UNITS: VOLT,
+    },
+]
+
+# Sensor data available to "Subaru Safety Plus" subscribers with PHEV vehicles
+EV_SENSORS = [
+    {
+        SENSOR_TYPE: "EV Range",
+        SENSOR_FIELD: sc.EV_DISTANCE_TO_EMPTY,
+        SENSOR_UNITS: LENGTH_MILES,
+    },
+    {
+        SENSOR_TYPE: "EV Battery Level",
+        SENSOR_FIELD: sc.EV_STATE_OF_CHARGE_PERCENT,
+        SENSOR_UNITS: PERCENTAGE,
+    },
+    {
+        SENSOR_TYPE: "EV Time to Full Charge",
+        SENSOR_FIELD: sc.EV_TIME_TO_FULLY_CHARGED,
+        SENSOR_UNITS: TIME_MINUTES,
+    },
+]
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Subaru sensors by config_entry."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id][ENTRY_COORDINATOR]
+    vehicle_info = hass.data[DOMAIN][config_entry.entry_id][ENTRY_VEHICLES]
+    entities = []
+    for vin in vehicle_info.keys():
+        _create_sensor_entities(entities, vehicle_info[vin], coordinator)
+    async_add_entities(entities, True)
+
+
+def _create_sensor_entities(entities, vehicle_info, coordinator):
+    sensors_to_add = []
+    if vehicle_info[VEHICLE_HAS_SAFETY_SERVICE]:
+        sensors_to_add.extend(SAFETY_SENSORS)
+
+        if vehicle_info[VEHICLE_API_GEN] == API_GEN_2:
+            sensors_to_add.extend(API_GEN_2_SENSORS)
+
+        if vehicle_info[VEHICLE_HAS_EV]:
+            sensors_to_add.extend(EV_SENSORS)
+
+    for subaru_sensor in sensors_to_add:
+        entities.append(
+            SubaruSensor(
+                vehicle_info,
+                coordinator,
+                subaru_sensor[SENSOR_TYPE],
+                subaru_sensor[SENSOR_FIELD],
+                subaru_sensor[SENSOR_UNITS],
+            )
+        )
+
+
+class SubaruSensor(SubaruEntity):
+    """Class for Subaru sensors."""
+
+    def __init__(self, vehicle_info, coordinator, entity_type, data_field, api_unit):
+        """Initialize the sensor."""
+        super().__init__(vehicle_info, coordinator)
+        self.hass_type = "sensor"
+        self.current_value = None
+        self.entity_type = entity_type
+        self.data_field = data_field
+        self.api_unit = api_unit
+
+    @property
+    def icon(self):
+        """Return icon for sensor."""
+        if self.entity_type == "EV Battery Level" and self.coordinator.data.get(
```

Comment:
Reverse the conditions and return the super then outdent the block below

Reply:
It might make more sense to set a device class instead of an icon for some of these